openslides/core/static/js/core/projector.js

```diff
@@ -134,6 +160,65 @@ angular.module('OpenSlidesApp.core.projector', ['OpenSlidesApp.core'])
     }
 ])
 
+.controller('SlideSpeakerOverlayCtrl', [
+    '$scope',
+    'Motion',
+    'Assignment',
+    'Agenda',
+    'Customslide',
+    'Projector',
+    'User',
+    function($scope, Motion, Assignment, Agenda, Customslide, Projector, User) {
+        // Attention! Each object that is used here has to be dealt on server side.
+        // Add it to the coresponding get_requirements method of the ProjectorElement
+        // class.
+
+        User.findAll();
+        $scope.visible = $scope.element.visible;
+        $scope.displayedElement = function() {
+            var displayedElement = [];
+            for (var e in $scope.$parent.elements) {
+                var ee = $scope.$parent.elements[e];
+                if (ee.name == 'motions/motion') {
+                    displayedElement = ['motion', ee.id];
+                } else if (ee.name == 'core/customslide') {
+                    displayedElement = ['agenda', ee.id];
+                } else if (ee.name == 'assignments/assignment') {
+                    displayedElement = ['assignment', ee.id];
+                }
+            }
+            if (displayedElement[0] == 'motion') {
```

Comment:
Hm. The motion was resolved in the slidesProvider so why do we have to find it again?